packages/story-editor/src/components/library/panes/text/stylePresets/stylePresets.js

```diff
@@ -0,0 +1,251 @@
+  const pushUpdate = useCallback(
+    (update) => {
+      updateSelectedElements({
+        properties: (element) => {
+          const updates = updateProperties(element, update, true);
+          const sizeUpdates = getUpdatedSizeAndPosition({
+            ...element,
+            ...updates,
+          });
+          return {
+            ...updates,
+            ...sizeUpdates,
+          };
+        },
+      });
+    },
+    [updateSelectedElements]
+  );
+
+  const handleApplyStyle = useApplyStyle({ pushUpdate });
+  const { addGlobalPreset } = useAddPreset({ presetType: PRESET_TYPES.STYLE });
+
+  if (!isStylePaneEnabled) {
+    return null;
+  }
+
+  const addStyledText = (preset) => {
+    insertElement(TYPE, {
+      ...DEFAULT_PRESET,
+      ...preset,
+    });
+  };
+
+  const handlePresetClick = (preset) => {
+    if (isText) {
+      handleApplyStyle(preset);
+    } else {
+      addStyledText(preset);
+    }
+  };
+
+  // If it's not text, we're displaying insertion overlay and label for adding a new text.
+  const styleItemProps = isText
+    ? {}
+    : {
+        activeItemOverlay: <InsertionOverlay />,
+        applyLabel: __('Add new text', 'web-stories'),
+      };
+
+  return (
+    <SectionContainer>
+      <TitleBar>
+        <Headline
+          as="h2"
+          size={THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.XXX_SMALL}
+        >
+          {__('Saved Styles', 'web-stories')}
+        </Headline>
+        <StyledButton
+          type={BUTTON_TYPES.TERTIARY}
+          size={BUTTON_SIZES.SMALL}
+          variant={BUTTON_VARIANTS.SQUARE}
+          onClick={addGlobalPreset}
+          aria-label={__('Add style', 'web-stories')}
+          disabled={!isText}
+        >
+          <Icons.Plus />
+        </StyledButton>
+      </TitleBar>
+      {hasPresets && (
+        <>
+          <StylesWrapper ref={stylesRef}>
+            {/* We only show the insertion overlay if we're inserting -> this means selected elements are not texts */}
+            <StyleGroup
+              styles={[...textStyles].reverse().slice(0, 2)}
+              handleClick={handlePresetClick}
+              buttonWidth={STYLE_BUTTON_WIDTH}
+              {...styleItemProps}
+            />
+          </StylesWrapper>
+          <StyledMoreButton
+            ref={buttonRef}
+            type={BUTTON_TYPES.PLAIN}
+            size={BUTTON_SIZES.SMALL}
+            variant={BUTTON_VARIANTS.RECTANGLE}
+            onClick={() => setIsPopupOpen(true)}
+          >
+            {__('More styles', 'web-stories')}
+            <Icons.ChevronDownSmall />
+          </StyledMoreButton>
+          <Popup
+            topOffset={topOffset}
+            isRTL={isRTL}
+            anchor={buttonRef}
+            isOpen={isPopupOpen}
+            placement={PLACEMENT.RIGHT_START}
+            spacing={SPACING}
+            renderContents={() => (
+              <StyleManager
+                styles={textStyles}
+                applyStyle={handlePresetClick}
```

Comment:
I feel like this should also close the fly-out on click. Was there specific UX on this?

Reply:
Hm, on applying it was not supposed to close in the panel, and the idea was to have the same logic here. However, when inserting a new, it should really close, will add it.